subsys/bluetooth/controller/ll_sw/ull_adv_iso.c

```diff
@@ -307,55 +501,131 @@ static uint32_t ull_adv_iso_start(struct ll_adv_iso *adv_iso,
 		ticks_slot_overhead = 0;
 	}
 
+	/* setup to use ISO create prepare function for first radio event */
+	mfy_lll_prepare.fp = lll_adv_iso_create_prepare;
+
 	/* TODO: Calculate ISO interval */
 	/* iso_interval shall be at least SDU interval,
 	 * or integer multiple of SDU interval for unframed PDUs
 	 */
-	iso_interval_us = adv_iso->sdu_interval;
+	iso_interval_us = adv_iso->lll.sdu_interval;
 
 	ret_cb = TICKER_STATUS_BUSY;
 	ret = ticker_start(TICKER_INSTANCE_ID_CTLR, TICKER_USER_ID_ULL_HIGH,
-			   (TICKER_ID_ADV_ISO_BASE + adv_iso->bis_handle),
+			   (TICKER_ID_ADV_ISO_BASE + adv_iso->lll.handle),
 			   ticks_anchor, 0,
 			   HAL_TICKER_US_TO_TICKS(iso_interval_us),
 			   HAL_TICKER_REMAINDER(iso_interval_us),
 			   TICKER_NULL_LAZY,
-			   (ll_adv_iso->evt.ticks_slot + ticks_slot_overhead),
-			   ticker_cb, ll_adv_iso,
+			   (adv_iso->evt.ticks_slot + ticks_slot_overhead),
+			   ticker_cb, adv_iso,
 			   ull_ticker_status_give, (void *)&ret_cb);
 	ret = ull_ticker_status_take(ret, &ret_cb);
 
-	return 0;
+	return ret;
 }
 
-static inline struct ll_adv_iso *ull_adv_iso_get(uint8_t handle)
+static void mfy_iso_offset_get(void *param)
 {
-	if (handle >= CONFIG_BT_CTLR_ADV_SET) {
-		return NULL;
-	}
+	struct ll_adv_sync_set *sync;
+	struct lll_adv_iso *lll_iso;
+	uint32_t ticks_to_expire;
+	struct pdu_big_info *bi;
+	uint32_t ticks_current;
+	struct pdu_adv *pdu;
+	uint8_t ticker_id;
+	uint8_t retry;
+	uint8_t id;
+
+	sync = param;
+	lll_iso = sync->lll.iso;
+	ticker_id = TICKER_ID_ADV_ISO_BASE + lll_iso->handle;
+
+	id = TICKER_NULL;
+	ticks_to_expire = 0U;
+	ticks_current = 0U;
+	retry = 4U;
+	do {
+		uint32_t volatile ret_cb;
+		uint32_t ticks_previous;
+		uint32_t ret;
+
+		ticks_previous = ticks_current;
+
+		ret_cb = TICKER_STATUS_BUSY;
+		ret = ticker_next_slot_get(TICKER_INSTANCE_ID_CTLR,
+					   TICKER_USER_ID_ULL_LOW,
+					   &id,
+					   &ticks_current, &ticks_to_expire,
+					   ticker_op_cb, (void *)&ret_cb);
+		if (ret == TICKER_STATUS_BUSY) {
+			while (ret_cb == TICKER_STATUS_BUSY) {
```

Comment:
Some clarifying comment needed: Polling on ret_cb, which is an output of a prior function.

Reply:
Added comments.